### PR TITLE
Backport/15129/stable14

### PR DESCRIPTION
**Pull request by rullzer**
Backport of #15129

### CODE REVIEW

File: lib/private/Repair/RemoveLinkShares.php [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+		return false;
+	}
+
+	/**
+	 * Delete the share
+	 *
+	 * @param int $id
+	 */
+	private function deleteShare(int $id): void {
```
Comment (nickvergessen) [inline lib/private/Repair/RemoveLinkShares.php]: :boom:

**Review by nickvergessen — DISMISSED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
🐘

File: lib/private/Repair/RemoveLinkShares.php [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+		if (!isset($this->userToNotify[$uid])) {
+			$this->userToNotify[$uid] = true;
+		}
+	}
+
+	/**
+	 * Send all notifications
+	 */
+	private function sendNotification() {
+		$time = $this->timeFactory->getDateTime();
```
Comment (nickvergessen) [inline lib/private/Repair/RemoveLinkShares.php]: le booooom :boom: #15563

Comment (MorrisJobke) [reply lib/private/Repair/RemoveLinkShares.php]: Fix is in #15570

**Pull request merged**